modules/fluid_properties/src/userobjects/BrineFluidProperties.C

```diff
@@ -35,7 +35,8 @@ BrineFluidProperties::BrineFluidProperties(const InputParameters & parameters)
   {
     const std::string class_name = "Water97FluidProperties";
     InputParameters params = _app.getFactory().getValidParams(class_name);
-    _fe_problem.addUserObject(class_name, water_name, params);
+    if (!_fe_problem.hasUserObject(water_name)) // multi-threaded guard
+      _fe_problem.addUserObject(class_name, water_name, params);
```

Comment:
Couldn't this still run in parallel causing a race condition?  If two threads run through the if condition check before either of them reach the addUserObject call - you end up the with the same problem you had before (although it is less likely to occur).

Reply:
Or maybe not - since object creation doesn't actually run in parallel.  This should probably be fine.

Reply:
`FEProblemBase::addUserObject` loops through threads. It is better here to use `if (_tid == 0)` though.